cucumber-expressions/java/src/main/java/io/cucumber/cucumberexpressions/TreeRegexp.java

```diff
@@ -41,18 +39,13 @@ private static GroupBuilder createGroupBuilder(Pattern pattern) {
             } else if (c == ']' && !escaping) {
                 charClass = false;
             } else if (c == '(' && !escaping && !charClass) {
-                groupStartStack.push(i);
-                boolean nonCapturing = isNonCapturingGroup(source, i);
-                GroupBuilder groupBuilder = new GroupBuilder();
-                if (nonCapturing) {
-                    groupBuilder.setNonCapturing();
-                }
```

Comment:
Why was this section deleted?

Reply:
With the adjusted GroupBuilder this is no longer necessary (all required data is passed in the constructor)